### PR TITLE
RavenDB-23445 Fixed indexing of null values for vector search

### PR DESCRIPTION
**Pull request by Lwiel**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-23445/Vector-Search-null-handling-for-string

### Additional description

Fixed indexing of null values for textual vector search

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [x] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs [hunk 1]
```diff
@@ -560,7 +560,13 @@ private static object VectorFromText(IndexField indexField, object value)
             {
                 var vectorList = new List<VectorValue>();
                 foreach (var item in enumerable)
+                {
+                    if (item is null)
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs]: Would it make sense to index this as "null" instead of skipping it?
If yes, maybe we should have some configuration option..

Comment (Lwiel) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs]: Changed

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs [hunk 2]
```diff
@@ -560,7 +560,10 @@ private static object VectorFromText(IndexField indexField, object value)
             {
                 var vectorList = new List<VectorValue>();
                 foreach (var item in enumerable)
+                {
```
Comment (maciejaszyk) [inline src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:684]: What if the `item` is `null`? It would throw an exception now. Should we return a list like `[vec0, vec2, VectorValue.Null, [...], vecN]` instead?

**Review by arekpalinski — APPROVED** [review]

**Comment by maciejaszyk** [comment]
Can you make this more generic and also add tests for null values for numerical vector search?

**Review by ppekrol — APPROVED** [review]

**Comment by maciejaszyk** [comment]
Please rebase the branch.

**Review by maciejaszyk — APPROVED** [review]
